atlas/AtlasMl/atlasml/ml/feedback_loop.py

```
@@ -0,0 +1,60 @@
+import numpy as np
+
+def update_cluster_centroid(
+        old_centroid: np.ndarray,
+        cluster_size: int,
+        new_point: np.ndarray
+) -> np.ndarray:
+    """
+    Incrementally updates a cluster centroid using the running-mean formula.
+    Args:
+        old_centroid (np.ndarray): Current centroid vector of shape (d,).
+        cluster_size (int): Number of points currently in the cluster (before adding new_point).
```

Ruff (0.12.2)

Line too long (96 > 88)

(E501)